### PR TITLE
AC: Replace use of os.path with pathlib, fix im2latex metric calculation on Linux

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py [hunk 1]
```diff
@@ -154,33 +155,33 @@ def render_routine(line):
...
     pre_name = os.path.normcase(output_path).replace('/', '_').replace('.', '_')
     formula = preprocess_formula(formula)
-    if not os.path.exists(output_path):
-        tex_filename = pre_name + '.tex'
-        log_filename = pre_name + '.log'
-        aux_filename = pre_name + '.aux'
-        with open(tex_filename, "w") as w:
+    if not output_path.exists():
+        tex_filename = Path(folder_path, pre_name + '.tex')
+        log_filename = tex_filename.with_name(pre_name + '.log')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py:163]:
```suggestion
        log_filename = tex_filename.with_suffix('.log')
```

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py [hunk 2]
```diff
@@ -154,33 +155,33 @@ def render_routine(line):
...
+        aux_filename = tex_filename.with_name(pre_name + '.aux')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py:164]:
```suggestion
        aux_filename = tex_filename.with_suffix('.aux')
```

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py [hunk 3]
```diff
@@ -154,33 +155,33 @@ def render_routine(line):
...
+        with open(str(tex_filename), "w") as w:
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py:165]:
```suggestion
        with tex_filename.open(mode="w") as w:
```

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py [hunk 4]
```diff
@@ -154,33 +155,33 @@ def render_routine(line):
...
         for filename in (tex_filename, log_filename, aux_filename):
-            if os.path.exists(filename):
-                os.remove(filename)
-        pdf_filename = tex_filename[:-4] + '.pdf'
-        png_filename = tex_filename[:-4] + '.png'
-        if not os.path.exists(pdf_filename):
+            if filename.exists():
+                filename.unlink()
+        pdf_filename = tex_filename.with_name(pre_name + '.pdf')
+        png_filename = tex_filename.with_name(pre_name + '.png')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py:173]:
```suggestion
        pdf_filename = tex_filename.with_suffix('.pdf')
        png_filename = tex_filename.with_suffix('.png')
```

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py [hunk 5]
```diff
@@ -154,33 +155,33 @@ def render_routine(line):
         line (tuple): formula idx, formula string, path to store rendered image
     """
     formula, file_idx, folder_path = line
-    output_path = os.path.join(folder_path, file_idx)
+    output_path = Path(folder_path, file_idx)
```
[the remark this answers is not in the document]
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py [hunk 6]
```diff
@@ -379,13 +380,13 @@ def render_images(self, annotations, predictions, images_dir):
...
+        out_path_gold = Path(images_dir, 'images_gold')
+        out_path_pred = Path(images_dir, 'images_pred')
         for dir_ in [out_path_gold, out_path_pred]:
-            if not os.path.exists(dir_):
-                os.makedirs(dir_)
-        lines_gold = [(ann.label, ann.identifier, out_path_gold) for ann in annotations]
-        lines_pred = [(pred.label, pred.identifier, out_path_pred) for pred in predictions]
+            if not dir_.exists():
+                dir_.mkdir()
+        lines_gold = [(ann.label, ann.identifier, str(out_path_gold)) for ann in annotations]
```
[the remark this answers is not in the document]
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py]: Removed

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py (hunk 1 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py (hunk 2 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py (hunk 3 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py (hunk 4 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py]: Done

**Review by eaidova — APPROVED** [review]

**Pull request merged**